File: poo/2. User Registration at a Gym JS/calculateTotal.java
public abstract class CalculateTotal {
    private final String name;
    private final String lastName;

    public CalculateTotal(String name, String lastName) {
        this.name = name;
        this.lastName = lastName;
    }

    public String getName() {
        return name;
    }

    public String getLastName() {
        return lastName;
    }

    public abstract double calculate();
}

File: poo/2. User Registration at a Gym/calculateTotal.php
<?php
    abstract class CalculateTotal  {
        readonly private ?string $name;
        readonly private ?string $lastName;

        public function __construct(string $name, string $lastName) {
            $this->name = $name;
            $this->lastName = $lastName;
        }

        public function getName(): string {
            return $this->name;
        }
        public function getLastName(): string {
            return $this->lastName;
        }
        
        abstract public function calculate(): float;
    }
?>